### DISCUSSION TITLE
Themes

### DISCUSSION THREAD

**Discussion by PMLS3** [unocss/unocss#629] [Q&A] [answered]
Is it possible to have more than one theme with different attributes? 

For instance: 

Theme a: {
colors: { primary: #0000}
}

Theme b: {
colors: { primary: #00FF00}
}

And the be able to select between the two

**Comment by chu121su12** [comment]
There's method in generator to set new config. You should be able to set new config with the alternate theme with it

https://github.com/unocss/unocss/blob/eca3d7ac6242a90560f28493fbe5200d8c0cba14/packages/core/src/generator/index.ts#L20

**Comment by luvletterldl** [comment, marked as answer]
@PMLS3 You can use [usecolormode](https://vueuse.org/core/usecolormode/#config) to resolve it.
some code for you:

Your css:
```css
:root[theme="dim"] {
  --primary:  0, 0, 0;
}
:root[theme="cafe"] {
  --primary: 0, 255, 0;
}
```

your unocss.config.ts
```typescript
rules: [
    [/^text-(.*)$/, ([, c], { theme }) => {
      if (theme.colors[c]) {
        return {
          color: theme.colors[c],
        }
      }
    }],
],
theme: {
    colors: {
      primary: 'rgb(var(--primary))',
    }
}
```

your .vue file
```vue
<script setup lang="ts">
import { useColorMode } from '@vueuse/core'

const mode = useColorMode({
  attribute: 'theme',
  modes: {
    // custom colors
    dim: 'dim',
    cafe: 'cafe',
  },
})

// if you want to change your theme, you just change model.value
// tips: It will produce a side effect, it will set 'vueuse-color-scheme' at your page's localStorage for record your theme.
mode.value = 'dim'
</script>

<template>
  <header class="text-primary">
    header
  </header>
  <button @click="mode = 'cafe'">switchToCafeTheme</button>
</template>
```

**Reply by PMLS3** [reply]
Much appreciated!

**Reply by jasonlam-swatow** [reply]
This way how to use opacity on colors? Like `text-primary/9`?

**Reply by zaynetro** [reply]
To support opacity you need to specify an alpha channel placeholder:

```js
theme: {
  colors: {
    primary: 'rgba(var(--my-color), %alpha)',
  }
}
```

Ref: https://github.com/unocss/unocss/blob/0f3b33b7a1dd4dda17340781139501c03b90982e/packages/preset-mini/src/_utils/colors.ts#L8
